src/Sylius/Bundle/CoreBundle/Applicator/ActionBasedDiscountApplicator.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\CoreBundle\Applicator;
+
+use Sylius\Bundle\CoreBundle\Calculator\CatalogPromotionPriceCalculatorInterface;
+use Sylius\Bundle\CoreBundle\DiscountApplicationCriteria\DiscountApplicationCriteriaInterface;
+use Sylius\Component\Core\Exception\ActionBasedPriceCalculatorNotFoundException;
+use Sylius\Component\Core\Model\CatalogPromotionInterface;
+use Sylius\Component\Core\Model\ChannelPricingInterface;
+use Sylius\Component\Promotion\Model\CatalogPromotionActionInterface;
+
+final class ActionBasedDiscountApplicator implements ActionBasedDiscountApplicatorInterface
+{
+    private CatalogPromotionPriceCalculatorInterface $priceCalculator;
+
+    private iterable $discountApplicatorCriteria;
+
+    public function __construct(
+        CatalogPromotionPriceCalculatorInterface $priceCalculator,
+        iterable $discountApplicatorCriteria
+    ) {
+        $this->priceCalculator = $priceCalculator;
+        $this->discountApplicatorCriteria = $discountApplicatorCriteria;
+    }
```

Comment:
Maybe we can already try

```php
public function __construct(
    private CatalogPromotionPriceCalculatorInterface $priceCalculator,
    private iterable $discountApplicatorCriteria
) {
}
```
? 🚀

Reply:
Code consistency question here: Will Sylius move to those promoted properties ?

Reply:
Good question and the answer is: I would like so 🖖 I've already started https://github.com/Sylius/Sylius/pull/13502